### ISSUE TITLE
Terminology Issue with Saving Models in trainer.py

### ISSUE THREAD

**Issue by HarryLTS** [pnnl/neuromancer#109]
When trainer.py calls self.logger.log_artifacts, "best_model_state_dict.pth" is a deepcopy of the state dict at the time of the model's best dev performance, but "best_model.pth" is the pickled version of self.model at the end of training, not at the best dev loss. This could be confusing to new users of the library.
